components/api/src/main/java/com/hotels/styx/api/HttpRequest.java

```diff
@@ -614,13 +619,78 @@ public Builder cookies(RequestCookie... cookies) {
          * @param cookies cookies
          * @return this builder
          */
-        public Builder cookies(List<RequestCookie> cookies) {
+        public Builder cookies(Collection<RequestCookie> cookies) {
             if (!cookies.isEmpty()) {
                 header(COOKIE, encode(cookies));
             }
             return this;
         }
 
+        /**
+         * Adds cookies into this request by overwriting the value of the "Cookie" header.
+         * <p>
+         * Note that this requires decoding the current header value before re-encoding, so it is most efficient to
+         * add all new cookies in one call to the method rather than spreading them out.
+         *
+         * @param cookies new cookies
+         * @return this builder
+         */
+        public Builder addCookies(RequestCookie... cookies) {
+            return addCookies(asList(cookies));
+        }
+
+        /**
+         * Adds cookies into this request by overwriting the value of the "Cookie" header.
```

Comment:
I cannot see any unit tests for `addCookies`. Please add tests for `addCookies()`, or am I just missing something?

Reply:
Please add unit tests for the following `addTests` scenarios:
1) add a new cookie when Cookie header is not present.
2) add a new cookie when Cookie header is already present.
3) modify a cookie value using `addCookie` when a Cookie header is present and the cookie is already set.

Reply:
Yes, there are missing tests. I will add tests for the new methods.